lib/compat/wordpress-6.0/class-wp-webfonts.php

```diff
@@ -291,4 +300,24 @@ public function generate_styles() {
 
 		return $styles;
 	}
+
+	/**
+	 * Preload webfonts to improve performance
+	 *
+	 * @return void
+	 */
+	public function preload_webfonts() {
+		foreach ( $this->get_fonts() as $font ) {
+			if ( $font['preload'] ) {
+				foreach( $font['src'] as $src ) {
+					$srcAsArray = explode( '.', $src );
+					$file_type = end( $srcAsArray );
+					if ( ! in_array( $file_type, self::$allowed_types ) ) {
+						$file_type = '';
+					}
+					echo '<link rel="preload" href="' . $src . '" as="font" type="font/' . $file_type . '" crossorigin />';
```

Comment:
The `WP_Webfonts_Provider_Local::order_src()` method returns an array of URLs along with their file-types. Perhaps we should use that? It would make sense, especially if we move the implementation to the `WP_Webfonts_Provider_Local` object (where this should be IMO)

Reply:
>The WP_Webfonts_Provider_Local::order_src() method returns an array of URLs along with their file/mime types. Perhaps we should use that?

Thinking about it, yes, this is a better approach to avoid redundancy.

I agree that this method is a better fit for the local provider (`WP_Webfonts_Provider_Local`) than here. In the future when Font Factory providers are bundled, this code will need changes as `src` is specific to locally hosted fonts.

Reply:
As this is rendering the HTML, the variables require escaping for security:

```suggestion
					printf(
						'<link rel="preload" href="%s" as="font" type="font/%s" crossorigin />',
						esc_url( $src ),
						esc_attr( $file_type )
					);
```